### PR TITLE
feat: move share consts from celestia-core

### PR DESCRIPTION
**Pull request by rootulp**
Part of https://github.com/celestiaorg/celestia-core/issues/841
Follow up to https://github.com/celestiaorg/celestia-core/pull/861

### CODE REVIEW

File: pkg/appconsts/appconsts.go [hunk 1]
```diff
@@ -3,10 +3,72 @@ package appconsts
...
+	// MaxShareVersion is the maximum value a share version can be.
+	MaxShareVersion = 127
+)
+
+var (
+	// IntermediateStateRootsNamespaceID is the namespace reserved for
+	// intermediate state root data
+	// TODO(liamsi): code commented out but kept intentionally.
+	// IntermediateStateRootsNamespaceID = namespace.ID{0, 0, 0, 0, 0, 0, 0, 2}
+
```
Comment (evan-forbes) [inline pkg/appconsts/appconsts.go:54]: [nit][blocking question]
wdyt about forwarding a link here to the missing constants? This way people don't have to remember which ones are where

like 
```go
TxNamespaceID = consts.TxNamespaceID
```

Comment (evan-forbes) [reply pkg/appconsts/appconsts.go:54]: if not, I'm also ok with approving as is. I think we're the only ones that use them

Comment (rootulp) [reply pkg/appconsts/appconsts.go:54]: Great idea, will do for all consts still in celestia-core

Comment (rootulp) [reply pkg/appconsts/appconsts.go]: Resolved in https://github.com/celestiaorg/celestia-app/pull/709/commits/6dde9c0f9d6ae56eda9eed6f5c28b3a1cb44e586

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**